src/utils/context.rs

```diff
@@ -22,19 +23,23 @@ impl<'a, 'r> Context<'a, 'r> {
         self.get::<ServiceCache>()
     }
 
-    pub fn host(&self) -> Option<String> {
-        self.request.headers().get_one("Host").map(|host| {
-            format!("{}://{}", scheme(), host)
-        })
-    }
-
     pub fn path(&self) -> String {
         self.request.uri().path().to_string()
     }
 
     pub fn uri(&self) -> String {
         self.request.uri().to_string()
     }
+
+    pub fn build_paging_link(&self, origin: Origin) -> String {
```

Comment:
I would also change this name to build_absolute_url or build_service_link ... Just more general vasically